### PR TITLE
Bugfix update connect vraserver function

### PR DESCRIPTION
**Pull request by Stevio54**
This pull request is related here: https://github.com/jakkulabs/PowervRA/issues/233

### CODE REVIEW

File: src/Functions/Public/Connect-vRAServer.ps1 [hunk 1]
```diff
@@ -85,162 +99,170 @@
...
 
-        }
-        catch [Exception]{
+            }
+            catch [Exception]{
+
+                if ($_.Exception.Response.StatusCode -eq "BadRequest") {
+                    # This could be due to an incorrectly set UserAttribute, so customizing the message.
+                    if ($Username -match "@") {
+                        Write-Host -ForegroundColor White -BackgroundColor DarkGray "Unfortunately, you have received a 400 BAD Request Error. Is it possible that your environment has userPrincipalName set as the Username attribute? If so, please make sure to include the parameter UserAttribute and set it to either UPN or userPrincipalName (e.g. -UserAttribute UPN)"
```
Comment (martin9700) [inline src/Functions/Public/Connect-vRAServer.ps1:256]: This should all be part of the throw message, or (my preference) Write-Error with an -ErrorAction Stop.  By using Write-Host it's now impossible for an upstream script to capture the error.

File: src/Functions/Public/Connect-vRAServer.ps1 [hunk 2]
```diff
@@ -85,162 +99,170 @@
...
+                    } else {
+                        Write-Host -ForegroundColor White -BackgroundColor DarkGray "Unfortunately, you have received a 400 BAD Request Error. It is possible that the Username you provided, $Username, is missing a domain (e.g. $Username@domain.com)?"
```
Comment (martin9700) [inline src/Functions/Public/Connect-vRAServer.ps1:258]: Make this the Throw message (or Write-Error).

File: src/Functions/Public/Connect-vRAServer.ps1 (hunk 1 above)
Comment (Stevio54) [reply src/Functions/Public/Connect-vRAServer.ps1:256]: Good note, update incoming.

File: src/Functions/Public/Connect-vRAServer.ps1 (hunk 2 above)
Comment (Stevio54) [reply src/Functions/Public/Connect-vRAServer.ps1:258]: good note, update incoming.

File: src/Functions/Public/Connect-vRAServer.ps1 (hunk 1 above)
Comment (Stevio54) [reply src/Functions/Public/Connect-vRAServer.ps1]: updated here: [new commit](https://github.com/jakkulabs/PowervRA/pull/235/commits/c6c50a2d5cb03f45234dc7e67e4c3ea169acf543)

File: src/Functions/Public/Connect-vRAServer.ps1 (hunk 2 above)
Comment (Stevio54) [reply src/Functions/Public/Connect-vRAServer.ps1]: updated here: [new commit](https://github.com/jakkulabs/PowervRA/pull/235/commits/c6c50a2d5cb03f45234dc7e67e4c3ea169acf543)

File: src/Functions/Public/Connect-vRAServer.ps1 [hunk 3]
```diff
@@ -12,6 +12,7 @@
         .PARAMETER Username
         Username to connect with
         For domain accounts ensure to specify the Username in the format username@domain, not Domain\Username
+        Note: UPN's are valid login Usernames as well and may also be in the format username@domain
```
Comment (jonathanmedd) [inline src/Functions/Public/Connect-vRAServer.ps1:15]:
```suggestion
        Note: UPNs are valid login Usernames as well and may also be in the format username@domain
```

Comment (Stevio54) [reply src/Functions/Public/Connect-vRAServer.ps1]: Just submitted the fix for this typo in latest commit to this pull request.

**Pull request merged**